internal/api/identity.go

```diff
@@ -109,6 +112,34 @@ func (a *API) linkIdentityToUser(ctx context.Context, tx *storage.Connection, us
 	if _, terr := a.createNewIdentity(tx, targetUser, providerType, structs.Map(userData.Metadata)); terr != nil {
 		return nil, terr
 	}
+
+	if targetUser.GetEmail() == "" {
+		if terr := targetUser.UpdateUserEmail(tx); terr != nil {
```

Comment:
Why is this? Above it the email seems to be empty, so why are we updating it to be empty?